src/HeaderSort/components/HeaderSortContainer.ts

```diff
@@ -0,0 +1,157 @@
+    listViewAvailable: boolean;
+    publishedSortAttribute?: string;
+    publishedSortOrder?: SortOrder;
+    publishedSortWidgetFriendlyId?: string;
+    targetListView?: ListView | null;
+    targetNode?: HTMLElement;
+}
+
+export default class HeaderSortContainer extends Component<ContainerProps, ContainerState> {
+    private navigationHandler: object;
+    private dataSourceHelper: DataSourceHelper;
+
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.state = { listViewAvailable: false };
+
+        this.updateSort = this.updateSort.bind(this);
+        this.navigationHandler = dojoConnect.connect(props.mxform, "onNavigation", this, this.connectToListView.bind(this));
+        this.subScribeToWidgetChanges = this.subScribeToWidgetChanges.bind(this);
+        this.publishWidgetChanges = this.publishWidgetChanges.bind(this);
+    }
+
+    render() {
+        return createElement("div", {
+                className: classNames("widget-header-sort", this.props.class),
+                style: SharedUtils.parseStyle(this.props.style)
+            },
+            createElement(Alert, {
+                bootstrapStyle: "danger",
+                className: "widget-header-sort-alert",
+                message: this.state.alertMessage
+            }),
+            this.renderSort()
+        );
+    }
+
+    componentDidUpdate(_prevProps: ContainerProps, prevState: ContainerState) {
+        if (this.state.listViewAvailable && !prevState.listViewAvailable) {
+            if (this.props.initialSorted) {
+                this.updateSort(this.props.sortAttribute, this.props.sortOrder);
+            }
+        }
+    }
+
+    componentWillUnmount() {
+        dojoConnect.disconnect(this.navigationHandler);
+    }
+
+    private renderSort(): ReactElement<HeaderSortProps> | null {
+        if (!this.state.alertMessage) {
+            return createElement(HeaderSort, {
+                caption: this.props.caption,
+                friendlyId: this.props.friendlyId,
+                initialSorted: this.props.initialSorted,
+                onClickAction: this.updateSort,
+                publishedSortAttribute: this.state.publishedSortAttribute,
+                publishedSortOrder: this.state.publishedSortOrder,
+                publishedSortWidgetFriendlyId: this.state.publishedSortWidgetFriendlyId,
+                sortAttribute: this.props.sortAttribute,
+                sortOrder: this.initialSortOrder(this.props.initialSorted, this.props.sortOrder)
+            });
+        }
+
+        return null;
+    }
+
+    private connectToListView() {
+        const queryNode = findDOMNode(this).parentNode as HTMLElement;
+        const targetNode = SharedUtils.findTargetNode(queryNode) as HTMLElement;
+        let targetListView: ListView | null = null;
+        let errorMessage = "";
+
+        if (targetNode) {
+            targetListView = dijitRegistry.byNode(targetNode);
+            if (targetListView) {
```

Comment:
We should re-write this function in a loop if possible to comply with the requirement of the configurations XML
`Select the same entity as the list view that this widget should interact with. If multiple list views with this entity are found, the closest is used`